packages/diagrams/frontend/sirius-components-diagrams/src/contexts/StoreContext.tsx

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import React, { useRef } from 'react';
+import { Edge, Node, useReactFlow } from 'reactflow';
+import { EdgeData, NodeData } from '../renderer/DiagramRenderer.types';
+import { SetState, StoreContextValue } from './StoreContext.types';
+
+const defaultValue: StoreContextValue = {
```

Comment:
That seems like a sizable change in term of philosophy so I'd like more information on that:

- Why do we need to encapsulate the function `setNodes` and `setEdges` from Reactflow in `useRef`?-
- Why should we expose those ref directly and thus let everybody deal with `.current` on a potentially `null` object?
- The lifecycle seems weird, why initialize it in `DiagramRenderer` like that?
- Shouldn't we use this setters everywhere instead of the one we use today? I wouldn't mind but in the current state of this PR it looks like yet another way to do things with a weird hack using `useRef` with an unknown impact in my mind.

Reply:
I modified the implementation to remove useRef and use a context with a custom hook instead.

Reply:
I rather migrate to this setter in smaller increment, this PR migrate handleNodesChange functions to this setter.
The "old" setter is still used in useAdjustSize and useConnector where it seems that more changes are needed to no break the current behavior.